relay-event-schema/src/protocol/contexts/trace.rs

```diff
@@ -135,7 +135,7 @@ pub struct TraceContext {
     pub sampled: Annotated<bool>,
 
     /// Data of the trace's root span.
-    #[metastructure(pii = "maybe", skip_serialization = "empty")]
+    #[metastructure(pii = "maybe", skip_serialization = "null")]
```

Comment:
Should we ever skip here? I assume the Python SDK setting data to `None` would be serialized as `null` which we then skip incorrectly.

Reply:
Skipping `null` would be correct IMO, but a side effect of this change is that `data: {}` is not be skipped anymore.  I will add some more tests.